hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java

```diff
@@ -260,6 +263,18 @@ public void start() {
       omMetadataManager.start(configuration);
     } catch (IOException ioEx) {
       LOG.error("Error starting Recon OM Metadata Manager.", ioEx);
+    } catch (RuntimeException runtimeException) {
+      LOG.warn("Unexpected runtime error starting Recon OM Metadata Manager.", runtimeException);
+      LOG.warn("Trying to delete existing recon OM snapshot DB and fetch new one.");
+      metrics.incrNumSnapshotRequests();
+      LOG.info("Fetching full snapshot from Ozone Manager");
+      // Update local Recon OM DB to new snapshot.
+      try {
+        boolean success = updateReconOmDBWithNewSnapshot();
+        LOG.info("Fetched full new snapshot from Ozone Manager: {}", success);
```

Comment:
Could we improve this log message? It'll look like this right now: `Fetched full new snapshot from Ozone Manager: true/false`, right?
I think we should go with separate log messages based on `success`, something like: `Successfully fetched a full snapshot from Ozone Manager`/`Failed fetching a full snapshot from Ozone Manager`.

Reply:
Thanks @dombizita for reviewing the patch. IMO, this will un-necessarily increase the complexity with `if-else` , because writing similar two `LOG.info` statements doesn't add any value except the word `Successfully` or `Failed`, If we just  log the `success` flag, it will be more concise I feel. Pls let me know your opinion on my explanation.  If you still feel, pls let me know, I'll do the change as suggested.

Reply:
I don't think a simple if-else increases complexity in this case. I thought of this because when someone will debug Recon and look the logs, they might slip on/misunderstand this log message and only see that Recon already "fetched full new snapshot".
```
Fetched full new snapshot from Ozone Manager: false
``` 
I don't see this kind of logs in Ozone anywhere else, where you state something and a simple true/false in the end of the log determines the success of the statement. 
Also when you are searching in logs usually you are searching for "success"/"fail", not for "true"/"false". Maybe we should also consider changing the log level based on the success of the snapshot fetch. 
Let me know what you think of these ideas.

Reply:
Ok @dombizita understood your point.  Done the change as per your suggestion. Thanks for explanation.